packages/focusvisible/src/useFocusVisible.js

```diff
@@ -0,0 +1,255 @@
+    relativeDocument.addEventListener('touchmove', onInitialPointerMove);
+    relativeDocument.addEventListener('touchstart', onInitialPointerMove);
+    relativeDocument.addEventListener('touchend', onInitialPointerMove);
+  }, [onInitialPointerMove, relativeDocument]);
+
+  const removeInitialPointerMoveListeners = useCallback(() => {
+    relativeDocument.removeEventListener('mousemove', onInitialPointerMove);
+    relativeDocument.removeEventListener('mousedown', onInitialPointerMove);
+    relativeDocument.removeEventListener('mouseup', onInitialPointerMove);
+    relativeDocument.removeEventListener('pointermove', onInitialPointerMove);
+    relativeDocument.removeEventListener('pointerdown', onInitialPointerMove);
+    relativeDocument.removeEventListener('pointerup', onInitialPointerMove);
+    relativeDocument.removeEventListener('touchmove', onInitialPointerMove);
+    relativeDocument.removeEventListener('touchstart', onInitialPointerMove);
+    relativeDocument.removeEventListener('touchend', onInitialPointerMove);
+  }, [onInitialPointerMove, relativeDocument]);
+
+  /* Unable to mock visibilityState in JSDom environment */
+  /* istanbul ignore next */
+  const onVisibilityChange = useCallback(() => {
+    if (relativeDocument.visibilityState === 'hidden') {
+      if (hadFocusVisibleRecently.current) {
+        hadKeyboardEvent.current = true;
+      }
+    }
+  }, [relativeDocument]);
+
+  useEffect(() => {
+    const currentScope = scope && scope.current;
+
+    relativeDocument.addEventListener('keydown', onKeyDown, true);
+    relativeDocument.addEventListener('mousedown', onPointerDown, true);
+    relativeDocument.addEventListener('pointerdown', onPointerDown, true);
+    relativeDocument.addEventListener('touchstart', onPointerDown, true);
+    relativeDocument.addEventListener('visibilitychange', onVisibilityChange, true);
+
+    addInitialPointerMoveListeners();
+
+    currentScope && currentScope.addEventListener('focus', onFocus, true);
+    currentScope && currentScope.addEventListener('blur', onBlur, true);
+
+    return () => {
+      relativeDocument.removeEventListener('keydown', onKeyDown);
+      relativeDocument.removeEventListener('mousedown', onPointerDown);
+      relativeDocument.removeEventListener('pointerdown', onPointerDown);
+      relativeDocument.removeEventListener('touchstart', onPointerDown);
+      relativeDocument.removeEventListener('visibilityChange', onVisibilityChange);
+
+      removeInitialPointerMoveListeners();
+
+      currentScope && currentScope.removeEventListener('focus', onFocus);
+      currentScope && currentScope.removeEventListener('blur', onBlur);
+
+      clearTimeout(hadFocusVisibleRecentlyTimeout.current);
+    };
+  }, [
+    addInitialPointerMoveListeners,
+    onBlur,
+    onFocus,
+    onKeyDown,
+    onPointerDown,
+    onVisibilityChange,
+    relativeDocument,
+    removeInitialPointerMoveListeners,
+    scope
```

Comment:
A `ref` changing will never trigger a useEffect to run, see https://medium.com/@teh_builder/ref-objects-inside-useeffect-hooks-eb7c15198780

Reply:
That's what I thought, but the exhaustive dependencies eslint warning is being triggered regardless. If it doesn't have an affect I might just leave it in to avoid disabling that rule locally.